presto-elasticsearch/src/test/java/com/facebook/presto/elasticsearch/TestElasticsearchIntegrationSmokeTest.java

```diff
@@ -90,4 +94,29 @@ public void testDescribeTable()
                 .row("comment", "varchar", "", "").build();
         assertEquals(actualResult, expectedColumns, format("%s != %s", actualResult, expectedColumns));
     }
+
+    @Test
+    public void testNestedFields()
+    {
+        String indexName = "data";
+
+        Map<String, Object> source = new HashMap<>();
+        source.put("name", "nestfield");
+        source.put("fields.fielda", 32);
+        source.put("fields.fieldb", "valueb");
+
+        embeddedElasticsearchNode.getClient().prepareIndex(indexName, "doc").setSource(source).get();
+        embeddedElasticsearchNode.getClient().admin().indices().refresh(refreshRequest(indexName)).actionGet();
+
+        MaterializedResult actualColumns = computeActual("SELECT name, fields.fielda, fields.fieldb FROM nested.data").toTestTypes();
+        MaterializedResult.Builder builder = resultBuilder(getQueryRunner().getDefaultSession(), VARCHAR, BIGINT, VARCHAR);
+        for (MaterializedRow row : actualColumns.getMaterializedRows()) {
+            builder.row(row.getField(0), row.getField(1), row.getField(2));
+        }
+
+        MaterializedResult actualResult = builder.build();
+        builder = resultBuilder(getQueryRunner().getDefaultSession(), VARCHAR, BIGINT, VARCHAR);
```

Comment:
`BIGINT` looks wrong since the column is defined as integer in nested.data.json.
https://github.com/prestodb/presto/pull/12948/files#diff-4e0329725eb70ef7e5720bda1caca46cR20

How about simplifying these code?
```java
        MaterializedResult actualResult = computeActual("SELECT name, fields.fielda, fields.fieldb FROM nested.data").toTestTypes();
        MaterializedResult expectedResult = resultBuilder(getQueryRunner().getDefaultSession(), VARCHAR, INTEGER, VARCHAR)
                .row("nestfield", 32, "valueb")
                .build();
        assertEquals(actualResult, expectedResult);
```